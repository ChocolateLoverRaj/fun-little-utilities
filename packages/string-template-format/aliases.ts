export { formatInspector as inspect, formatInspector as dbg } from './format-inspect'
export { formatJson as json } from './format-json'
export { formatToString as toString, formatToString as str } from './format-to-string'
export { formatUri as uri, formatUriComponent as uriComp } from './format-uri'
