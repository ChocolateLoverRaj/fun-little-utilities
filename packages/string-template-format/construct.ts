import compose from './compose'
import Tag from './tag'

export interface Constructor<X, Y> {
  new (x: X): Y
}

export interface CallableConstructor<X, Y> {
  (x: X): Y
}

export const CallableConstructor =
  <X, Y> (cls: Constructor<X, Y>): CallableConstructor<X, Y> => x => new cls(x)

export const construct = <Fx, Gy, FyGx> (
  g: Constructor<FyGx, Gy>,
  f: Tag<Fx, FyGx>
): Tag<Fx, Gy> => compose(CallableConstructor(g), f)

export default construct
