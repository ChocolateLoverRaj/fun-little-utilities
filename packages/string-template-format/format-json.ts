import Tag from './tag'

/**
 * Convert values to JSON string
 */
export const formatJson = Tag(JSON.stringify)
export default formatJson
