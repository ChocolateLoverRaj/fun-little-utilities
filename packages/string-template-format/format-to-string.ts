import Tag from './tag'

/**
 * Type of objects that can be passed to `formatToString`
 */
export interface ToStringObject {
  readonly toString: () => string
}

/**
 * Type of values that can be passed to `formatToString`
 */
export type ToStringValue = ToStringObject | undefined | null

/**
 * Format any value that is convertible to string
 * (i.e. can be passed to `String`)
 */
export const formatToString = Tag(String as (raw: ToStringValue) => string)
export default formatToString
