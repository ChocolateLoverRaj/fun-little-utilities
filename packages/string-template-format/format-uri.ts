import Tag from './tag'

/**
 * Format string into valid URI
 */
export const formatUri = Tag(encodeURI)

/**
 * Format string into valid URI component
 */
export const formatUriComponent = Tag(encodeURIComponent)
