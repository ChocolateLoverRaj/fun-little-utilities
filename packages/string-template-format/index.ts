export * from './compose'
export * from './construct'
export * from './format-inspect'
export * from './format-json'
export * from './format-to-string'
export * from './tag'
export { default } from './tag'
