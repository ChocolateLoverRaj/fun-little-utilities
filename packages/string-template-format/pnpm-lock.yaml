dependencies:
  '@types/node': 12.7.11
  string-template-format-base: 'link:../string-template-format-base'
  string-template-format-compose: 'link:../string-template-format-compose'
  string-template-format-inspect: 'link:../string-template-format-inspect'
  string-template-format-json: 'link:../string-template-format-json'
  string-template-format-tostring: 'link:../string-template-format-tostring'
  string-template-format-uri: 'link:../string-template-format-uri'
  tslib: 1.10.0
lockfileVersion: 5.1
packages:
  /@types/node/12.7.11:
    dev: false
    resolution:
      integrity: sha512-Otxmr2rrZLKRYIybtdG/sgeO+tHY20GxeDjcGmUnmmlCWyEnv2a2x1ZXBo3BTec4OiTXMQCiazB8NMBf0iRlFw==
  /tslib/1.10.0:
    dev: false
    resolution:
      integrity: sha512-qOebF53frne81cf0S9B41ByenJ3/IuH8yJKngAX35CmiZySA0khhkovshKK+jGCaMnVomla7gVlIcc3EvKPbTQ==
specifiers:
  '@types/node': ^12.7.11
  string-template-format-base: ^0.1.0
  string-template-format-compose: ^0.1.0
  string-template-format-inspect: ^0.1.1
  string-template-format-json: ^0.1.0
  string-template-format-tostring: ^0.1.0
  string-template-format-uri: ^0.1.0
  tslib: ^1.10.0
