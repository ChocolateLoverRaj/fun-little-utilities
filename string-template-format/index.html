<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>string-template-format — Reference</title>
	<meta name="description" content="">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="assets/js/search.js" data-base=".">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="index.html" class="title">string-template-format — Reference</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="globals.html">Globals</a>
				</li>
			</ul>
			<h1> string-template-format — Reference</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<div class="tsd-panel tsd-typography">
				<a href="#string-template-format" id="string-template-format" style="color: inherit; text-decoration: none;">
					<h1>String Template Format</h1>
				</a>
				<p>String template tag that accepts any value and returns a string</p>
				<a href="#usage-examples" id="usage-examples" style="color: inherit; text-decoration: none;">
					<h2>Usage Examples</h2>
				</a>
				<a href="#embed-objects-into-string-as-json" id="embed-objects-into-string-as-json" style="color: inherit; text-decoration: none;">
					<h3>Embed objects into string as JSON</h3>
				</a>
				<pre><code class="language-typescript"><span class="hljs-keyword">import</span> { formatJson } <span class="hljs-keyword">from</span> <span class="hljs-string">'string-template-format'</span>
<span class="hljs-built_in">console</span>.log(formatJson<span class="hljs-string">`number <span class="hljs-subst">${<span class="hljs-number">123</span>}</span>; string: <span class="hljs-subst">${<span class="hljs-string">'abc'</span>}</span>; object: <span class="hljs-subst">${{ abc: <span class="hljs-number">123</span> }</span>}; array: <span class="hljs-subst">${[<span class="hljs-number">0</span>, <span class="hljs-number">1</span>, <span class="hljs-number">2</span>]}</span>`</span>)</code></pre>
				<p><em>should print:</em></p>
				<pre><code>number: <span class="hljs-number">123</span>; <span class="hljs-built_in">string</span>: <span class="hljs-string">"abc"</span>; object: {<span class="hljs-string">"abc"</span>:<span class="hljs-number">123</span>}; <span class="hljs-built_in">array</span>: [<span class="hljs-number">0</span>,<span class="hljs-number">1</span>,<span class="hljs-number">2</span>]</code></pre>
				<a href="#format-objects-with-codeutilinspectcode" id="format-objects-with-codeutilinspectcode" style="color: inherit; text-decoration: none;">
					<h3>Format objects with <code>util.inspect</code></h3>
				</a>
				<pre><code class="language-typescript"><span class="hljs-keyword">import</span> { formatInspect } <span class="hljs-keyword">from</span> <span class="hljs-string">'string-template-format'</span>
<span class="hljs-built_in">console</span>.log(formatInspect<span class="hljs-string">`number <span class="hljs-subst">${<span class="hljs-number">123</span>}</span>; string: <span class="hljs-subst">${<span class="hljs-string">'abc'</span>}</span>; object: <span class="hljs-subst">${{ abc: <span class="hljs-number">123</span> }</span>}; array: <span class="hljs-subst">${[<span class="hljs-number">0</span>, <span class="hljs-number">1</span>, <span class="hljs-number">2</span>]}</span>`</span>)</code></pre>
				<p><em>should print:</em></p>
				<pre><code>number: <span class="hljs-number">123</span>; <span class="hljs-built_in">string</span>: <span class="hljs-string">'abc'</span>; object: { abc: <span class="hljs-number">123</span> }; <span class="hljs-built_in">array</span>: [ <span class="hljs-number">0</span>, <span class="hljs-number">1</span>, <span class="hljs-number">2</span> ]</code></pre>
				<a href="#custom-tags" id="custom-tags" style="color: inherit; text-decoration: none;">
					<h3>Custom tags</h3>
				</a>
				<pre><code class="language-typescript"><span class="hljs-keyword">import</span> { tag } <span class="hljs-keyword">from</span> <span class="hljs-string">'string-template-format'</span>
<span class="hljs-keyword">const</span> myTag = tag(<span class="hljs-function"><span class="hljs-params">value</span> =&gt;</span> <span class="hljs-string">`[<span class="hljs-subst">${<span class="hljs-keyword">typeof</span> value}</span> <span class="hljs-subst">${value}</span>]`</span>)
<span class="hljs-built_in">console</span>.log(myTag<span class="hljs-string">`<span class="hljs-subst">${<span class="hljs-number">123</span>}</span>; <span class="hljs-subst">${<span class="hljs-string">'abc'</span>}</span>; <span class="hljs-subst">${{ abc: <span class="hljs-number">123</span> }</span>}; <span class="hljs-subst">${[<span class="hljs-number">0</span>, <span class="hljs-number">1</span>, <span class="hljs-number">2</span>]}</span>`</span>)</code></pre>
				<p><em>should print:</em></p>
				<pre><code>[number <span class="hljs-number">123</span>]; [<span class="hljs-built_in">string</span> abc]; [object [Object object]]; [<span class="hljs-built_in">array</span> <span class="hljs-number">0</span>,<span class="hljs-number">1</span>,<span class="hljs-number">2</span>]</code></pre>
				<a href="#license" id="license" style="color: inherit; text-decoration: none;">
					<h2>License</h2>
				</a>
				<p><a href="https://git.io/fxKXN">MIT</a> © <a href="https://github.com/KSXGitHub">Hoàng Văn Khải</a></p>
			</div>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals  ">
						<a href="globals.html"><em>Globals</em></a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/callableconstructor.html" class="tsd-kind-icon">Callable<wbr>Constructor</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/compose.html" class="tsd-kind-icon">Compose</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/constructor.html" class="tsd-kind-icon">Constructor</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/dumper.html" class="tsd-kind-icon">Dumper</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/formatter.html" class="tsd-kind-icon">Formatter</a>
					</li>
					<li class=" tsd-kind-interface tsd-is-external">
						<a href="interfaces/inspector.html" class="tsd-kind-icon">Inspector</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/loader.html" class="tsd-kind-icon">Loader</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/readonlyjsonarray.html" class="tsd-kind-icon">Readonly<wbr>Json<wbr>Array</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/readonlyjsonobject.html" class="tsd-kind-icon">Readonly<wbr>Json<wbr>Object</a>
					</li>
					<li class=" tsd-kind-interface tsd-is-external">
						<a href="interfaces/replacer.html" class="tsd-kind-icon">Replacer</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/reviver.html" class="tsd-kind-icon">Reviver</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/templatetag.html" class="tsd-kind-icon">Template<wbr>Tag</a>
					</li>
					<li class=" tsd-kind-interface tsd-is-external">
						<a href="interfaces/tostringobject.html" class="tsd-kind-icon">To<wbr>String<wbr>Object</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/writablejsonarray.html" class="tsd-kind-icon">Writable<wbr>Json<wbr>Array</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter tsd-is-external">
						<a href="interfaces/writablejsonobject.html" class="tsd-kind-icon">Writable<wbr>Json<wbr>Object</a>
					</li>
					<li class=" tsd-kind-type-alias tsd-is-external">
						<a href="globals.html#inspectformatter" class="tsd-kind-icon">Inspect<wbr>Formatter</a>
					</li>
					<li class=" tsd-kind-type-alias tsd-is-external">
						<a href="globals.html#jsonscalar" class="tsd-kind-icon">Json<wbr>Scalar</a>
					</li>
					<li class=" tsd-kind-type-alias tsd-is-external">
						<a href="globals.html#jsonvalue" class="tsd-kind-icon">Json<wbr>Value</a>
					</li>
					<li class=" tsd-kind-type-alias tsd-has-type-parameter tsd-is-external">
						<a href="globals.html#readonlyjsonvalue" class="tsd-kind-icon">Readonly<wbr>Json<wbr>Value</a>
					</li>
					<li class=" tsd-kind-type-alias tsd-is-external">
						<a href="globals.html#tostringvalue" class="tsd-kind-icon">To<wbr>String<wbr>Value</a>
					</li>
					<li class=" tsd-kind-type-alias tsd-has-type-parameter tsd-is-external">
						<a href="globals.html#writablejsonvalue" class="tsd-kind-icon">Writable<wbr>Json<wbr>Value</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#templatetag" class="tsd-kind-icon">Template<wbr>Tag</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#compose" class="tsd-kind-icon">compose</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#dump" class="tsd-kind-icon">dump</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#formatinspector" class="tsd-kind-icon">format<wbr>Inspector</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#formatjson" class="tsd-kind-icon">format<wbr>Json</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#formattostring" class="tsd-kind-icon">format<wbr>ToString</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#formaturi" class="tsd-kind-icon">format<wbr>Uri</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#formaturicomponent" class="tsd-kind-icon">format<wbr>Uri<wbr>Component</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#load" class="tsd-kind-icon">load</a>
					</li>
					<li class=" tsd-kind-function tsd-has-type-parameter tsd-is-external">
						<a href="globals.html#callableconstructor" class="tsd-kind-icon">Callable<wbr>Constructor</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#inspectformatter" class="tsd-kind-icon">Inspect<wbr>Formatter</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#inspector" class="tsd-kind-icon">Inspector</a>
					</li>
					<li class=" tsd-kind-function tsd-has-type-parameter tsd-is-external">
						<a href="globals.html#construct" class="tsd-kind-icon">construct</a>
					</li>
					<li class=" tsd-kind-function tsd-has-type-parameter tsd-is-external">
						<a href="globals.html#tag" class="tsd-kind-icon">tag</a>
					</li>
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-module"><span class="tsd-kind-icon">Module</span></li>
				<li class="tsd-kind-object-literal"><span class="tsd-kind-icon">Object literal</span></li>
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
				<li class="tsd-kind-function tsd-has-type-parameter"><span class="tsd-kind-icon">Function with type parameter</span></li>
				<li class="tsd-kind-index-signature"><span class="tsd-kind-icon">Index signature</span></li>
				<li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li>
				<li class="tsd-kind-type-alias tsd-has-type-parameter"><span class="tsd-kind-icon">Type alias with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-enum"><span class="tsd-kind-icon">Enumeration</span></li>
				<li class="tsd-kind-enum-member"><span class="tsd-kind-icon">Enumeration member</span></li>
				<li class="tsd-kind-property tsd-parent-kind-enum"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-enum"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
				<li class="tsd-kind-interface tsd-has-type-parameter"><span class="tsd-kind-icon">Interface with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-interface"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-interface"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-interface"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
				<li class="tsd-kind-class tsd-has-type-parameter"><span class="tsd-kind-icon">Class with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class"><span class="tsd-kind-icon">Accessor</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-class"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-call-signature tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="assets/js/main.js"></script>
<script>if (location.protocol == 'file:') document.write('<script src="assets/js/search.js"><' + '/script>');</script>
</body>
</html>