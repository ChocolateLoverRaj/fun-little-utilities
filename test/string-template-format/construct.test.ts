import { construct, formatInspector } from 'string-template-format'

it('Error', () => {
  expect(construct(Error, formatInspector)`
    number: ${123}
    string: ${'abc'}
    undefined: ${undefined}
    function: ${() => undefined}
    class: ${class MyClass {}}
    object: ${{ abc: 123, def: 456 }}
    array: ${[0, 1, 2, 3, 4]}
  `).toMatchSnapshot()
})

it('TypeError', () => {
  expect(construct(TypeError, formatInspector)`
    number: ${123}
    string: ${'abc'}
    undefined: ${undefined}
    function: ${() => undefined}
    class: ${class MyClass {}}
    object: ${{ abc: 123, def: 456 }}
    array: ${[0, 1, 2, 3, 4]}
  `).toMatchSnapshot()
})

it('RangeError', () => {
  expect(construct(RangeError, formatInspector)`
    number: ${123}
    string: ${'abc'}
    undefined: ${undefined}
    function: ${() => undefined}
    class: ${class MyClass {}}
    object: ${{ abc: 123, def: 456 }}
    array: ${[0, 1, 2, 3, 4]}
  `).toMatchSnapshot()
})

it('SyntaxError', () => {
  expect(construct(SyntaxError, formatInspector)`
    number: ${123}
    string: ${'abc'}
    undefined: ${undefined}
    function: ${() => undefined}
    class: ${class MyClass {}}
    object: ${{ abc: 123, def: 456 }}
    array: ${[0, 1, 2, 3, 4]}
  `).toMatchSnapshot()
})
