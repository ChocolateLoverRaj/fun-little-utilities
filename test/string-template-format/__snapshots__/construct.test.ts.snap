// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error 1`] = `
[Error: 
    number: 123
    string: 'abc'
    undefined: undefined
    function: [Function]
    class: [Function: MyClass]
    object: { abc: 123, def: 456 }
    array: [ 0, 1, 2, 3, 4 ]
  ]
`;

exports[`RangeError 1`] = `
[RangeError: 
    number: 123
    string: 'abc'
    undefined: undefined
    function: [Function]
    class: [Function: MyClass]
    object: { abc: 123, def: 456 }
    array: [ 0, 1, 2, 3, 4 ]
  ]
`;

exports[`SyntaxError 1`] = `
[SyntaxError: 
    number: 123
    string: 'abc'
    undefined: undefined
    function: [Function]
    class: [Function: MyClass]
    object: { abc: 123, def: 456 }
    array: [ 0, 1, 2, 3, 4 ]
  ]
`;

exports[`TypeError 1`] = `
[TypeError: 
    number: 123
    string: 'abc'
    undefined: undefined
    function: [Function]
    class: [Function: MyClass]
    object: { abc: 123, def: 456 }
    array: [ 0, 1, 2, 3, 4 ]
  ]
`;
